tests/snapshot_tests.cc

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <memory>
+#include <seastar/core/sstring.hh>
+#include <seastar/core/future-util.hh>
+#include <seastar/core/align.hh>
+#include <seastar/core/do_with.hh>
+#include <seastar/core/sleep.hh>
+#include <seastar/testing/test_case.hh>
+#include "sstables/sstables.hh"
+#include "sstable_utils.hh"
+#include "schema.hh"
+#include "compress.hh"
+#include "database.hh"
+#include "db/config.hh"
+#include "tmpdir.hh"
+#include "cql_test_env.hh"
+
+using namespace sstables;
+
+future<> do_with_some_data(std::function<future<> (cql_test_env&)> fut) {
+    return seastar::async([fut = std::move(fut)] () mutable {
+        db::config db_cfg;
+        db_cfg.data_file_directories(std::vector<sstring>({ tmpdir().path().string() }));
+        do_with_cql_env([fut = std::move(fut)] (cql_test_env& e) {
+            e.create_table([](sstring ks_name) {
```

Comment:
should use do_with_cql_env_thread(), it works by accident because you're already in a thread here, but it's not robust.